### PR TITLE
Make credentials compatible with HTTP Request Nodes

### PR DESCRIPTION
**Pull request by pascal-zarrad**
This PR fixes the credentials in the n8n Akeneo Node. With these changes:
 - the credentials can be used with HTTP Request Nodes
 - the credentials are working properly with the Akeneo Node
 - are not requested with every call

### CODE REVIEW

**Comment by pascal-zarrad** [comment]
Note:  Changes to `tsconfig.json` match official template: https://github.com/n8n-io/n8n-nodes-starter/blob/master/tsconfig.json

File: credentials/utilities/AkeneoApiAuthentication.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		username: credentials.username,
+		password: credentials.password,
+	};
+
+	const response = await http.helpers.httpRequest({
+		url: getAuthenticationUrl(credentials),
+		method: 'POST',
+		headers: {
+			'Content-Type': 'application/json',
+			Authorization: 'Basic ' + basicAuthCredentials,
```
Comment (wapec) [inline credentials/utilities/AkeneoApiAuthentication.ts]: not critical but a better way to write is 
```
Authorization: `Basic ${basicAuthCredentials}`,
```

File: credentials/utilities/AkeneoApiAuthentication.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		grant_type: 'refresh_token',
+		refresh_token: credentials.refresh_token,
+	};
+
+	const response = await http.helpers.httpRequest({
+		url: getAuthenticationUrl(credentials),
+		method: 'POST',
+		headers: {
+			'Content-Type': 'application/json',
+			Authorization: 'Basic ' + basicAuthCredentials,
```
Comment (wapec) [inline credentials/utilities/AkeneoApiAuthentication.ts]: same here
```
Authorization: `Basic ${basicAuthCredentials}`,
```

File: credentials/utilities/AkeneoApiAuthentication.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	credentials.token_lifetime = authResponse.expires_in;
+	credentials.always_expired = '';
+
+	return credentials;
+}
+
+function getAuthenticationUrl(credentials: ICredentialDataDecryptedObject): string {
+	if (typeof credentials.akeneo_base_url !== 'string') {
+		throw new Error(
+			'Type ' + typeof credentials.akeneo_base_url + ' is not a valid type for an URL',
```
Comment (wapec) [inline credentials/utilities/AkeneoApiAuthentication.ts]: same here
```
`Type ${typeof credentials.akeneo_base_url} is not a valid type for an URL`,
```

File: nodes/Akeneo/helpers/akeneoRequest.ts [hunk 4]
```diff
@@ -1,58 +1,94 @@
...
-	try{
-		const response = await axios.get(url, headers);
-		return response.data;
-	}catch(e){
-		return {error: e};
+const GET = async (nodeExecuteFunc: IAllExecuteFunctions, { url, headers }: typeAkeneoRequest) => {
+	try {
+		return nodeExecuteFunc.helpers.httpRequestWithAuthentication.call(
+			nodeExecuteFunc,
+			'akeneoApi',
```
Comment (wapec) [inline nodes/Akeneo/helpers/akeneoRequest.ts]: 'akeneoApi' is used several times later, better to have a const variable for it

File: credentials/utilities/AkeneoApiAuthentication.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+		refresh_token: response.refresh_token,
+		expires_in: response.expires_in,
+	};
+}
+
+export function updateCredentials(
+	credentials: ICredentialDataDecryptedObject,
+	authResponse: AkeneoAuthResponse,
+): ICredentialDataDecryptedObject {
+	credentials.access_token = authResponse.access_token;
```
Comment (wapec) [inline credentials/utilities/AkeneoApiAuthentication.ts]: there is a way to destructure the object by
```
const { access_token, refresh_token, expires_in } = authResponse;
```

File: credentials/utilities/AkeneoApiAuthentication.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	credentials.refresh_token = authResponse.refresh_token;
+	credentials.last_refresh = Date.now();
+	credentials.token_lifetime = authResponse.expires_in;
+	credentials.always_expired = '';
```
Comment (wapec) [inline credentials/utilities/AkeneoApiAuthentication.ts]: then you can assign credentials like
```
credentials = { 
    ...credentials,
    access_token,
    refresh_token,
    token_lifetime: expires_in,
    last_refresh: Date.now(),
    always_expired: ''	
}

**Review by wapec — APPROVED** [review]

**Pull request merged**